H2MLauncher.UI/Dialog/Views/PasswordDialog.xaml.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Media;
+
+using H2MLauncher.UI.Dialog;
+
+namespace H2MLauncher.UI
+{
+    public partial class PasswordDialog : UserControl
+    {
+        public PasswordDialog()
+        {
+            InitializeComponent();
+
+        }
+        private void OK_Click(object sender, RoutedEventArgs e)
+        {
+            if (this.DataContext is not PasswordViewModel viewModel)
+                return;
+
+            viewModel.Password = PasswordInput.Password;
```

Comment:
Is this the reason you used event handlers instead of command binding? (like not setting the password until it actually is confirmed) 

Otherwise, use bindings directly to the view model and check for the result of `ShowDialog`

Reply:
Yeah , i do so . So we only set the password if the user actually did press "Ok" button